### DISCUSSION TITLE
How can I generate a query for retreiving text of liked tweets of a list of users?

### DISCUSSION THREAD

**Discussion by camillaancona** [DocNow/twarc#689] [Q&A] [answered]
Hello, I would like to know if is it possible to use the following command: 
twarc2 liked-tweets [OPTIONS] USER_ID [OUTFILE] with a list of users rather than a single user_id at a time. 
If not, how can I make a simple python code to create a workaround? 
I have the same issue with the following command: twarc2 liking-users [OPTIONS] TWEET_ID [OUTFILE] specifying a list of tweets ids instead of a single one

Thanks a lot

**Comment by igorbrigadir** [comment, marked as answer]
Unfortunately the command line doesn't have an already working command to do this in bulk, but you can script it in bash

For example, if you have an `input.txt` with 1 user per line like this (it should work with either usernames or user IDs):

```
user1
user2
user3
```

This should get likes of each of them:

```bash
while read line; do twarc2 liked-tweets "$line" "likes_of_$line.jsonl" && echo $line; done < input.txt
```

This will create `likes_of_user1.jsonl`, `likes_of_user2.jsonl`, `likes_of_user3.jsonl`.

Alternatively, using https://twarc-project.readthedocs.io/en/latest/api/client2/#twarc.client2.Twarc2.liked_tweets is the function to call, but this way you'll have to manage your own way of writing output - the function is a generator that returns API response pages.

**Reply by camillaancona** [reply]
Thank you, the bash code for the following command twarc2 liking-users [OPTIONS] TWEET_ID [OUTFILE] is the same only changing the input file (tweets_ids instead of users_ids)?

**Reply by igorbrigadir** [reply]
yep, same way:

```bash
while read line; do twarc2 liking-users "$line" "users_who_liked_tweet_$line.jsonl" && echo $line; done < input.txt
```

if the `input.txt` is 1 tweet ID per line

**Reply by camillaancona** [reply]
Thanks a lot. I retrieved the list of tweets_ids usign the deyhadrate twarc2 command starting from my jsonl file d tweets (created from searches). Then I used the bash command line that you suggested as while read line; do twarc2 liking-users "$line" "users_who_liked_tweet_$line.jsonl" && echo $line; done < tweets-ids.txt
however it returns me an error saying that ] is not validary parameter value but the input text file contains only the tweets ids one per line. Can you help me?
[tweet-ids.txt](https://github.com/DocNow/twarc/files/10715692/tweet-ids.txt)

**Reply by igorbrigadir** [reply]
>  ] is not validary parameter

This is usually something wrong with where you're running the command - it should be inside a Bash terminal.
